packages/foam-vscode/src/features/commit-on-save.ts

```diff
@@ -0,0 +1,100 @@
+import { commands, ExtensionContext, TextDocument, workspace, extensions } from "vscode";
+import { GitExtension, Commit } from "../git";
+import { FoamFeature } from "../types";
+import { exec } from 'child_process';
+import { promisify } from 'util';
+
+const execP = promisify(exec);
+let latestCommitTime;
+let latestSyncTime;
+let commitInterval = 10; // replace with configuration
+let syncInterval = 60 * 5;
+
+const feature: FoamFeature = {
+  activate: (context: ExtensionContext) => {
+    context.subscriptions.push(
+      commands.registerCommand("foam-vscode.run-hook-on-save", commitOnSave)
+    );
+    workspace.onDidSaveTextDocument((document: TextDocument) => {
```

Comment:
I saw that in your list of todos, I imagine this would live behind a feature flag to enable/disable it?

Reply:
Yes